### PR TITLE
Add string option for vm.runIn*Context methods in Node.js typing

### PR DESCRIPTION
**Pull request by mohsen1**
This is part of my work at https://github.com/webpack/webpack/pull/6862
## Link to code 
https://github.com/nodejs/node/blob/77b52fd58f7398a81999c81afd21fe2e156c0766/lib/vm.js#L284-L286

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/node/index.d.ts [hunk 1]
```diff
@@ -2001,10 +2002,11 @@ declare module "vm" {
     }
     export function createContext(sandbox?: Context): Context;
     export function isContext(sandbox: Context): boolean;
-    export function runInContext(code: string, contextifiedSandbox: Context, options?: RunningScriptOptions): any;
+    export function runInContext(code: string, contextifiedSandbox: Context, options?: RunningScriptOptions | string): any;
+    /** @deprecated */
```
Comment (mohsen1) [inline types/node/index.d.ts]: https://github.com/nodejs/node/blob/de0053cc3280bdf72c9010f383290f79120a1e98/doc/api/deprecations.md#dep0069-vmrunindebugcontextstring

**Comment by Flarna** [comment]
Even this is in source code it's not in API documentation. I think this is NodeJS 0.10 API compatibility which had only string but no options object in API.
I'm not sure if we should rely on this undocumented compat code to be in code forever. It would be nice to add also a PR towards NodeJs doc. Depending on the result if this PR we would have a clear statement.

On the other hand it is in code now and also on master so for NodeJs 8 and 9 it will be for sure not removed....

**Comment by mohsen1** [comment]
The type definition is marking existing and working core as incorrect. The definition should reflect the API even if not documented

**Review by Flarna — CHANGES_REQUESTED** [review]
Could you please add the test also for v9?

**Comment by Flarna** [comment]
I still think that a PR towards NodeJS doc would be nice (like it was done for fs in https://github.com/nodejs/node/pull/1806) but this is not part of this PR.

**Review by Flarna — APPROVED** [review]

**Pull request merged**